model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -22,34 +22,38 @@
 from mo.front.common.partial_infer.utils import int64_array, float32_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
 from mo.graph.graph import Graph, rename_nodes
 from mo.middle.replacement import MiddleReplacementPattern
-from mo.ops.const import Const
+from mo.ops.reshape import Reshape
 from mo.ops.shape import Shape
 from mo.ops.strided_slice import StridedSlice
 
 
 class UnsqueezeTileReshapeBlockToInterpolate(MiddleReplacementPattern):
     """
-    This transformation looks for Interpolation layer implemented using simple operations, i.e. Unsqueeze,
-    Tile, Reshape, and replaces found pattern with a sequence of Shape, StridedSlice, Const, Mul, Interpolate.
-
-    Here we assume that the input of 'unsqueeze' is in NDHWC layout and is a 5D-tensor.
-
-    Found pattern will be replaced with
-        nodes=[
-            ('shape', dict(kind='op', op='Shape')),
-            ('strided_slice', dict(kind='op', op='StridedSlice')),
-            ('scales', dict(kind='op', op='Const')),
-            ('scaled_shape', dict(kind='op', op='Mul')),
-            ('interp', dict(kind='op', op='Interpolate'))
-        ],
-        edges=[
-            ('shape', 'strided_slice', {'in': 0}),
-            ('strided_slice', 'scaled_shape', {'in': 0}),
-            ('scales', 'scaled_shape', {'in': 1}),
-            ('scaled_shape', 'interp', {'in': 1}),
-        ]
+    The transformation looks for a sub-graph performing unsqueeze-ing input tensor by some "axis" and then tiling over
+    it fixed number of "times". This pattern can be represented with the Interpolate operation  of mode "nearest"
+    performing interpolation over specific "axis" with fixed output dimension size equal to "times".
+
+    Note, that the transformation expects that the output from Tile is reshaped back to the tensor with rank equal to
+    the input tensor rank. This constraints occurs because the pattern appears in the models where these patterns appear
+    one after another, performing unsqueeze-ing over different dimensions, effectively performing interpolation over
+    several dimensions.
+
+    These sequences are merged in the 'optimizer/extensions/middle/InterpolateSequenceToInterpolate.py' transformation
+    into a single Interpolate operation.
+
+    The transformation is applicable only when all following conditions are fulfilled:
+
+    1. 'Unsqueeze' must be performed with respect to only one axis.
+    2. The length of the value of the second input of 'Tile' must be equal to the input rank of 'Unsqueeze' plus 1.
+    3. All elements of the value of the second input of 'Tile' must be equal to 1,
+       except the value corresponding the interpolated axis.
+    4. The input rank of 'Unsqueeze' and the output rank of 'Reshape' must be equal.
+
+    Finally, because plugins support only Interpolate-4 with 4D or 5D tensor with interpolated data,
+    we need to check that the input rank of 'Unsqueeze' is equal to 4 or 5.
```

Comment:
Also, the idea of this transformation will be more clear if you share example of some input that is processed through a chain `Unqueeze->Tile->Reshape` that is equivalent to neighbor Interpolate. This example will be a bit of help to these many words.

Reply:
Done.